bin/strata-client/src/args.rs

```diff
@@ -74,65 +73,54 @@ impl Args {
     /// Overrides passed directly as args and not as overrides.
     fn get_direct_overrides(&self) -> Vec<String> {
         let mut overrides = Vec::new();
+        if self.sequencer {
+            overrides.push("client.is_sequencer=true".to_string());
+        }
         if let Some(datadir) = &self.datadir {
-            overrides.push(format!("client.datadir={}", datadir.to_string_lossy()));
+            overrides.push(format!("client.datadir={datadir:?}"));
         }
         if let Some(rpc_host) = &self.rpc_host {
-            overrides.push(format!("client.rpc_host={}", rpc_host));
+            overrides.push(format!("client.rpc_host={rpc_host}"));
         }
         if let Some(rpc_port) = &self.rpc_port {
-            overrides.push(format!("client.rpc_port={}", rpc_port));
+            overrides.push(format!("client.rpc_port={rpc_port}"));
         }
 
         overrides
     }
 }
 
-type Override = (Vec<String>, String);
+type Override = (String, toml::Value);
 
 /// Parses an overrides This first splits the string by '=' to get key and value and then splits
 /// the key by '.' which is the update path.
 pub fn parse_override(override_str: &str) -> Result<Override, ConfigError> {
-    let (key, value) = override_str
+    let (key, value_str) = override_str
         .split_once("=")
         .ok_or(ConfigError::InvalidOverride(override_str.to_string()))?;
-    let path: Vec<_> = key.split(".").map(|x| x.to_string()).collect();
-    Ok((path, value.to_string()))
-}
-
-pub fn apply_overrides(overrides: Vec<String>, table: &mut Table) -> Result<Config, ConfigError> {
-    for res in overrides.iter().map(String::as_str).map(parse_override) {
-        let (path, val) = res?;
-        apply_override(&path, &val, table)?;
-    }
-
-    toml::Value::Table(table.clone())
-        .try_into()
-        .map_err(|_| ConfigError::ConfigNotParseable)
+    Ok((key.to_string(), parse_value(value_str)))
 }
 
 /// Apply override to config.
 pub fn apply_override(
-    path: &[String],
-    str_value: &str,
+    path: &str,
+    value: toml::Value,
     table: &mut Table,
 ) -> Result<(), ConfigError> {
-    match path {
-        [key] => {
-            let value = parse_value(str_value);
-            table.insert(key.to_string(), value);
+    match path.split_once(".") {
+        None => {
+            table.insert(path.to_string(), value);
             Ok(())
         }
-        [key, other @ ..] => {
+        Some((key, rest)) => {
             if let Some(t) = table.get_mut(key).and_then(|v| v.as_table_mut()) {
-                apply_override(other, str_value, t)
+                apply_override(rest, value, t)
             } else if table.contains_key(key) {
                 Err(ConfigError::TraversePrimitiveAt(key.to_string()))
```

Comment:
Actually I know this is what I named it in the code I sent but it's not strictly correct since it might be an array.  Maybe `TraverseNonTableAt`?  I'm not sure.  Maybe this doesn't matter.

Reply:
Yes, I too think that `TraverseNonTableAt` is more precise.